pkg/config/labels/instance.go

```diff
@@ -92,14 +92,20 @@ func (i Instance) Validate() error {
 
 // IsDNS1123Label tests for a string that conforms to the definition of a label in
 // DNS (RFC 1123).
-func IsDNS1123Label(value string) bool {
-	return len(value) <= DNS1123LabelMaxLength && dns1123LabelRegexp.MatchString(value)
+func IsDNS1123Label(value string) (bool, error) {
```

Comment:
This is awkward. If you need to return an error, consider just changing this to a `Check` style method:

```golang
func CheckDNS1123Label(value string) error {
...
}
```

Call sites that are just looking for true/false will then just check the error against `nil`.